photon-client/src/integTest/scala/com/linkedin/photon/ml/cli/game/training/DriverTest.scala

```diff
@@ -315,42 +406,80 @@ class DriverTest extends SparkTestUtils with GameTestUtils with TestTemplateWith
   }
 
   /**
+   * Check that we can calculate feature shard statistics correctly.
+   */
+  @Test
+  def testCalculateFeatureShardStats() : Unit = sparkTest("calculateFeatureShardStats", useKryo = true) {
+
+    val outputDir = s"$getTmpDir/output"
+    val indexMapPath = getClass.getClassLoader.getResource("GameIntegTest/input/feature-indexes").getPath
+    val args = CommonTestUtils.argArray(
+      fixedAndRandomEffectSeriousRunArgs() ++
+        Map(
+          "output-dir" -> outputDir,
+          "offheap-indexmap-dir" -> indexMapPath,
+          "offheap-indexmap-num-partitions" -> "1",
+          "summarization-output-dir" -> outputDir))
+
+    val trainingRecordsPath = getClass.getClassLoader.getResource("GameIntegTest/input/train").getPath
+    val params = GameParams.parseFromCommandLine(args)
+    val logger = new PhotonLogger(s"${params.outputDir}/log", sc)
+    val driver = new Driver(sc, params, logger)
+    val featureIndexMapLoaders = driver.prepareFeatureMaps()
+    val data: DataFrame =
+      new AvroDataReader(sc)
+        .readMerged(
+          trainingRecordsPath,
+          featureIndexMapLoaders,
+          params.featureShardIdToFeatureSectionKeysMap,
+          numPartitions = 1)
+
+    val S = driver.calculateAndSaveFeatureShardStats(data, featureIndexMapLoaders)
+    assertTrue(S.isDefined)
+    val stats = S.get.toMap
+    assertEquals(stats.size, featureIndexMapLoaders.size)
+    featureIndexMapLoaders.keys.foreach { featureShardId => assertTrue(stats(featureShardId).mean.length > 0) }
```

Comment:
Ideally we would check that the summary matches pre-computed "correct" values, though I'm not sure if it makes sense to do here (since `Driver` is not responsible for computing summaries correctly, just that summaries are computed).

However, this test should check that the summaries are saved to the correct directory